azurerm/resource_arm_key_vault_access_policy_test.go

```diff
@@ -0,0 +1,135 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMKeyVaultAccessPolicy(t *testing.T) {
+	resourceName := "azurerm_key_vault.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMKeyVaultAccessPolicy(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "access_policy.0.key_permissions.0", "create"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMKeyVaultAccessPolicy_update(t *testing.T) {
+	ri := acctest.RandInt()
+	resourceName := "azurerm_key_vault.test"
+	preConfig := testAccAzureRMKeyVaultAccessPolicy(ri, testLocation())
+	postConfig := testAccAzureRMKeyVaultAccessPolicy_update(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMKeyVaultDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "access_policy.0.key_permissions.0", "create"),
+					resource.TestCheckResourceAttr(resourceName, "access_policy.0.secret_permissions.0", "set"),
+					resource.TestCheckResourceAttr(resourceName, "tags.environment", "Production"),
+				),
+			},
+			{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "access_policy.0.key_permissions.0", "get"),
+					resource.TestCheckResourceAttr(resourceName, "access_policy.0.secret_permissions.0", "get"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMKeyVaultAccessPolicyDestroy(s *terraform.State) error {
```

Comment:
This should be used to ensure that resources setup by your acceptance tests are detroyed.